holoviews/plotting/bokeh/raster.py

```diff
@@ -245,13 +245,28 @@ def _get_cmapper_opts(self, low, high, factors, colors):
 
     def _get_colormapper(self, eldim, element, ranges, style, factors=None,
                          colors=None, group=None, name='color_mapper'):
+        dict_cmap = False
+        if isinstance(style.get("cmap"), dict):
+            dict_cmap = style["cmap"]
+            style["cmap"] = list(dict_cmap.values())
```

Comment:
Logically it makes more sense to validate for missing value dimensions before setting this (though for readability I can see why you might prefer this order)

Reply:
If we don't convert the `style["cmap"]` to a list the colormapper will be of type `CategoricalColorMapper`, and `WeightedStackColorMapper` only accepts a `ContinuousColorMapper`. So the conversion to a list is to get the allowed colormapper.

Reply:
Thanks for the explanation. I thought it would be something like that but it is good to confirm.